index.js

```diff
@@ -23,42 +23,62 @@ module.exports = {
 		this._app = appOrAddon.app || appOrAddon;
 		this.options = new Config(this._app, this.ui);
 
-		this.importPolyfills();
+		this.importPolyfill();
 	},
 
-	importPolyfills() {
-		const { polyfillBundle, buildForProduction } = this.options;
+	importPolyfill() {
+		const { polyfillBundle } = this.options;
 
-		const webcomponentsjsPath = path.join(this._app.bowerDirectory, 'webcomponentsjs');
-		const customElementsEs5Adapter = path.join(webcomponentsjsPath, 'custom-elements-es5-adapter.js');
-		const webcomponentsjsPolyfill = path.join(webcomponentsjsPath, `webcomponents-${polyfillBundle}.js`);
+		if (polyfillBundle && polyfillBundle !== 'none') {
+			const webcomponentsjsPath = path.join(this._app.bowerDirectory, 'webcomponentsjs');
+			const webcomponentsjsPolyfill = path.join(webcomponentsjsPath, `webcomponents-${polyfillBundle}.js`);
 
-		// Import custom-elements-es5-adapter if ES6 Classes are transpiled to ES5
-		// for browsers that natively support Custom Elements
-		// https://github.com/webcomponents/webcomponentsjs#custom-elements-es5-adapterjs
-		if (buildForProduction.enabled && buildForProduction.build.js.compile) {
-			this._app.import(customElementsEs5Adapter, { options: 'prepend' });
+			this._app.import(webcomponentsjsPolyfill, { options: 'prepend' });
 		}
-
-		this._app.import(webcomponentsjsPolyfill, { options: 'prepend' });
 	},
 
 	// insert polyfills and bundled elements
 	contentFor(type, config) {
-		if (type !== 'head') {
-			return null;
-		}
+		if (type === 'head-footer') {
+			const headContents = [];
+			const { globalPolymerSettings, buildForProduction } = this.options;
 
-		const headContents = [];
-		const { globalPolymerSettings } = this.options;
+			if (globalPolymerSettings) {
+				headContents.push(`<script>window.Polymer = ${JSON.stringify(globalPolymerSettings)};</script>`);
+			}
 
-		if (globalPolymerSettings) {
-			headContents.push(`<script>window.Polymer = ${JSON.stringify(globalPolymerSettings)};</script>`);
+			if (buildForProduction.enabled && buildForProduction.build.js.compile) {
+				headContents.push(this.getCustomEs5Adapter());
+			}
+
+			return headContents.join('\n');
+		} else if (type === 'body-footer') {
+			return this.getBundleImport(config);
 		}
 
-		headContents.push(this.getBundleImport(config));
+		return null;
+	},
 
-		return headContents.join('\n');
+	getCustomEs5Adapter() {
+		const webcomponentsPolyfillsPath = path.join('assets', this._app.bowerDirectory, 'webcomponentsjs');
+
+		/**
+		 * Include custom-elements-es5-adapter only for browsers that natively support customElements
+		 * Using Rob Dodson solution: https://youtu.be/Ucq9F-7Xp8I?t=7m57s
+		 * https://github.com/webcomponents/webcomponentsjs/tree/v1#custom-elements-es5-adapterjs
+		 */
+		const customEs5Adapter = `
+		<div id="ce-es5-shim">
+			<script>
+				if (!window.customElements) {
```

Comment:
You should never inject javascript code in the main html file without babelifiying it.

```js
const babel = require('@babel/core');

function babelify(filePath, config) {
	const { code } = babel.transformFileSync(filePath, config);

	return code;
}

// addon context
{
	_getPresets() {
		const targets = this.project && this.project.targets;
		const presetOptions = Object.assign({}, {
			modules: false,
			targets
		});

		return [
			[require.resolve('@babel/preset-env'), presetOptions]
		];
	},

	_getBabelConfig() {
		return {
			presets: this._getPresets()
		};
	},



	contentFor(type, config) {
		const jsCode = babelify('js_rel_path', this._getBabelConfig());
	},
}
```

Reply:
Check the addon `lib/app-shell/index.js` file (inside our project) for more info

Reply:
custom-elements-es5-adapter [should not be transpiled](https://github.com/webcomponents/webcomponentsjs#custom-elements-es5-adapterjs). Check out this issue to understand the motivation behind it: https://github.com/webcomponents/webcomponentsjs/issues/749

And this is [the reason why this polyfill is used](https://github.com/Polymer/tools/blob/master/packages/build/src/js-transform.ts#L31-L36) instead of `@babel/plugin-transform-classes`